### PR TITLE
`ConstScoreQuery`

### PR DESCRIPTION
**Pull request by shikhar**
Based this on `BoostQuery`, was able to reuse the existing `ConstScorer`.

### CODE REVIEW

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Can we move the ConstScorer in there too?

File: src/query/const_score_query.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+/// The document set matched by the `ConstScoreQuery` is strictly the same as the underlying query.
+/// The configured score is used for each document.
+pub struct ConstScoreQuery {
+    query: Box<dyn Query>,
+    score: Score,
+}
+
+impl ConstScoreQuery {
+    /// Builds a const score query.
+    pub fn new(query: Box<dyn Query>, score: Score) -> ConstScoreQuery {
```
Comment (fulmicoton) [inline src/query/const_score_query.rs:19]: I think the `ConstScoreQuery` symbol is not accessible from client in the library.
(hence the clippy warning below)

Can you re-export it from `tantivy::query`?

File: src/query/const_score_query.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    fn explain(&self, reader: &SegmentReader, doc: u32) -> crate::Result<Explanation> {
+        let mut scorer = self.scorer(reader, 1.0)?;
+        if scorer.seek(doc) != doc {
+            return Err(TantivyError::InvalidArgument(format!(
+                "Document #({}) does not match",
+                doc
+            )));
+        }
+        let mut explanation =
```
Comment (shikhar) [inline src/query/const_score_query.rs]: Should we avoid calling the child scorer here too?

```rust
let mut explanation = Explanation::new("Const", self.score);
```

Comment (fulmicoton) [reply src/query/const_score_query.rs]: Either way works for me.

**Review by fulmicoton — CHANGES_REQUESTED** [review]
please fix clippy

File: src/query/const_score_query.rs (hunk 2 above)
Comment (shikhar) [reply src/query/const_score_query.rs]: Removing the inner score from here, it gets included in detail anyway

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**